### PR TITLE
Small fixes

### PR DESCRIPTION
**Pull request by lachrist**
Mainly a fix for https://github.com/applandinc/appmap-agent-js/issues/112 and https://github.com/applandinc/appmap-agent-js/issues/113. I also did some refactoring and changed to scope of formatting and linting.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,10 +18,10 @@
 ### Requirements
 
 Supported platforms:
-* Unix-like os, tested on Linux and macOS; Windows is currently not supported
-* Node.js 14, 16 or 18
-* Express.js
-* git is highly recommended 
+* Mainstream os: linux distros, osx, and windows
```
Comment (apotterri) [inline README.md:21]: Why not leave this as "macOS", i.e. the official name Apple uses?

File: components/hook-http/node/index.mjs [hunk 2]
```diff
@@ -1,14 +1,16 @@
 const {
   Buffer: { from: toBuffer, concat: concatBuffer },
-  Object: { fromEntries },
+  Object: { fromEntries, entries: toEntries },
   TextDecoder,
   Reflect: { apply },
   JSON: { parse: parseJSON },
+  Math: { round },
   undefined,
 } = globalThis;
```
Comment (apotterri) [inline components/hook-http/node/index.mjs:9]: For my own edification: I understand the renaming that's happening here, but why are you destructuring `TextDecoder` and `undefined` from `globalThis`? Can't you just reference in the code without qualification?

**Review by apotterri — APPROVED** [review]
I had a couple of questions, but changes all look good.

File: components/hook-http/node/index.mjs (hunk 2 above)
Comment (lachrist) [reply components/hook-http/node/index.mjs:9]: This to protect against the user changing the global object. It is to separate as much as possible the observed application layer from the agent layer.

File: README.md (hunk 1 above)
Comment (lachrist) [reply README.md]: I don't know why I changed it, I revert it 36718ff399386ac0feec8b889e65044c8c3d15af

**Pull request merged**